test/common/index.js

```diff
@@ -690,12 +690,26 @@ Object.defineProperty(exports, 'hasIntl', {
 });
 
 // Useful for testing expected internal/error objects
+const kAcceptableErrorTypes = ['NodeError', 'AssertionError', 'RangeError'];
 exports.expectsError = function expectsError({code, type, message}) {
-  return function(error) {
+  const expectedTypeName = type && type.name;
+  return function expectsErrorCheck(error) {
     assert.strictEqual(error.code, code);
-    if (type !== undefined)
-      assert(error instanceof type,
-             `${error} is not the expected type ${type}`);
+    if (expectedTypeName) {
```

Comment:
> or a subclass of TypeError

That is the problem. In a recent PR, someone used `common.expectsError()` with `Error`, when they meant `TypeError`. The test should have failed on this.

In our test suite we almost exclusively want the strictest error checks possible. Even if we were to add `strictType` (which I don't think we should), how can we achieve that without exposing the fact that we're using the internal errors?